components/TokenSelectionModal.tsx

```diff
@@ -63,18 +79,149 @@ const TokenSelectionItem = ({ token, onClick }: TokenSelectionItemProps) => {
+      borderRadius=".5rem"
+      padding=".5rem"
+    >
+      <Text fontSize="md">No results</Text>
+    </Flex>
+  );
+};
+
+const CustomTokenSelectionItem = ({
+  onSelect,
+  key,
+  tokenAddress,
+}: CustomTokenSelectionItemProps) => {
+  const { notifyUser } = useNotifications();
+  const { chain } = useNetwork();
+  const { isOpen: isCustomOpen, onOpen: onCustomOpen, onClose: onCustomClose } = useDisclosure();
+  const {
+    isOpen: isBlacklistOpen,
+    onOpen: onBlacklistOpen,
+    onClose: onBlacklistClose,
+  } = useDisclosure();
+  const network = networks[chain?.id || 1];
+  const tokenLink = `${network.blockExplorerUrl}token/${tokenAddress}`;
+  const { data, isError, isLoading } = useWagmiToken({ address: tokenAddress });
+  const { address } = useAccount();
+  const {
+    data: balanceData,
+    isError: isBalanceError,
+    isLoading: isBalanceLoading,
+  } = useBalance({
+    address,
+    token: tokenAddress,
+    chainId: chain?.id,
+  });
+  const isBlacklisted = network.tokenBlacklist.get(tokenAddress);
+  const openModal = isBlacklisted ? onBlacklistOpen : onCustomOpen;
+
+  if (isError) {
+    notifyUser({
+      alertType: 'warning',
+      message: 'Failed to fetch custom token',
+    });
+  }
+
+  if (isBalanceError) {
+    notifyUser({
+      alertType: 'error',
+      message: 'Failed to fetch custom token balance',
+    });
+  }
+
+  if (isLoading) {
+    return <Spinner />;
+  }
+
+  if (data) {
+    const token = { ...data, logoURI: '', chainId: chain!.id, balance: balanceData?.value || null };
+    return (
+      <>
+        <TokenSelectionItem
+          token={token}
+          key={key}
+          onClick={openModal}
+          isBalanceLoading={isBalanceLoading}
+        />
+        <AlertModal
+          isOpen={isCustomOpen}
+          onClose={onCustomClose}
+          onClick={() => {
+            onSelect(token);
+            onCustomClose();
+          }}
+          address={tokenAddress}
+          description={`This token isn't included in the railgun token list. Always conduct your own research before shielding.`}
+        >
+          <Flex my=".75rem" justify="center">
+            <Flex
+              alignItems="center"
+              gap=".30rem"
+              padding=".50rem"
+              bg="gray.100"
+              borderRadius="1rem"
+              _hover={{ backgroundColor: 'rgba(184, 192, 220, 0.15)' }}
+            >
+              <Link href={tokenLink} _hover={{ textDecoration: 'none' }} isExternal maxW="18rem">
+                <Text
+                  cursor="pointer"
+                  maxW="20rem"
+                  textOverflow="ellipsis"
+                  overflow="hidden"
+                  whiteSpace="nowrap"
+                >
+                  {tokenLink}
+                </Text>
+              </Link>
+              <Copy text={tokenLink}>
+                <CopyIcon cursor="pointer" />
+              </Copy>
```

Comment:
We could probably get rid of the copy functionality/component to simplify things and just keep the "open in Etherscan" part. Since there isn't much of a use case for copying the Etherscan link and it's not a patter I've really seen before